metacat-common-server/src/main/java/com/netflix/metacat/common/server/events/MetacatEventBus.java

```diff
@@ -79,22 +77,29 @@ public void postAsync(final ApplicationEvent event) {
     public void postSync(final ApplicationEvent event) {
         log.debug("Received request to post an event {} synchronously", event);
         this.eventSyncCounter.increment();
-        this.eventPublisher.publishEvent(event);
+        super.multicastEvent(event);
     }
 
-    /**
-     * Registers an object.
-     *
-     * @param object object
-     */
-    public void register(final Object object) {
+    @Override
+    public void addApplicationListener(final ApplicationListener listener) {
```

Comment:
So instead of the extra code with annotations like we saw in that example today instead what you're doing here is registering all listeners in both classes and then based on the API called publishing to one or the other. Are you sure the `super.multicastEvent(event)` call is going to use the default synchronous task executor? I just worry this is a house of cards relying on too much assumptions in configuration that we could screw up at any time or other starters could rather than making it explicit